mesonbuild/build.py

```diff
@@ -62,6 +63,73 @@ class DFeatures(TypedDict):
         import_dirs: T.List[IncludeDirs]
         versions: T.List[T.Union[str, int]]
 
+    class BuildTargetKeywordArguments(TypedDict, total=False):
+
+        # The use of Sequence[str] here is intentional to get the generally
+        # undesirable behavior of Sequence[str]
```

Comment:
What undesirable behavior?

Reply:
That plain `str` is a valid `Sequence[str]`. What Meson almost always actually want is more like `list[str] | tuple[str, ...] | ...` but *not* plain `str`

Reply:
I see... It seems it could technically be possible to have as SequenceNotStr protocol: https://stackoverflow.com/a/79310749

Reply:
I tried this at one point and I recall that it completely didn't work, but I forget the details. Hopefully you can figure it out but I'm not optimistic.

Reply:
I also tried it several times as well and couldn't make it work either. But if someone else has actually figured it out, and interested to try. There's some places in Build specifically that would really benefit from it if we can make it work.